### PR TITLE
Generating allowed lists in Markdown

### PR DESCRIPTION
**Pull request by vigejolla**

### CODE REVIEW

**Comment by Thaodan** [comment]
Should the *.conf files be in the tree if the are generated during build?

**Comment by vigejolla** [comment]
> Should the *.conf files be in the tree if the are generated during build?

They are not generated during build

File: allowed_libraries.conf [hunk 1]
```diff
@@ -1,7 +1,9 @@
 # Pattern matching is supported as withing BASH shell with 'extglob' enabled.
 # https://www.gnu.org/software/bash/manual/html_node/Pattern-Matching.html
+#
+# First line of comment block is used as header in documentation
```
Comment (sledges) [inline allowed_libraries.conf:4]: Should this comment be also in other .conf files where applicable?

File: allowed_libraries.conf [hunk 2]
```diff
@@ -1,7 +1,9 @@
 # Pattern matching is supported as withing BASH shell with 'extglob' enabled.
```
Comment (sledges) [inline allowed_libraries.conf:1]:
```suggestion
# Pattern matching is supported as within BASH shell with 'extglob' enabled.
```
Looks like an old typo:)

**Review by sledges — CHANGES_REQUESTED** [review]
`->` looks like a sneaked-in typo: https://github.com/sailfishos/sdk-harbour-rpmvalidator/blob/7125b42afbf81817cc77d2545ec3363fa8b8bba7/allowed_qmlimports.conf#L38

File: allowed_libraries.conf (hunk 1 above)
Comment (vigejolla) [reply allowed_libraries.conf:4]: Probably :)

Comment (vigejolla) [reply allowed_libraries.conf:4]: Added

File: allowed_libraries.conf (hunk 2 above)
Comment (vigejolla) [reply allowed_libraries.conf]: fixed

**Comment by vigejolla** [comment]
> `->` looks like a sneaked-in typo:
> 
> https://github.com/sailfishos/sdk-harbour-rpmvalidator/blob/7125b42afbf81817cc77d2545ec3363fa8b8bba7/allowed_qmlimports.conf#L38

That was sneaky!

**Comment by vigejolla** [comment]
> > `->` looks like a sneaked-in typo:
> > https://github.com/sailfishos/sdk-harbour-rpmvalidator/blob/7125b42afbf81817cc77d2545ec3363fa8b8bba7/allowed_qmlimports.conf#L38
> 
> That was sneaky!

fixed now as well

**Review by sledges — APPROVED** [review]

File: doc/base.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  - ${lib}
+% endfor
+
+% endif
+</%def>
+<%def name="makerenamelist(section, subsection)">\
+% if len(subsection) > 1:
+% for lib in subsection[1:]:
+  - ${lib}
+    - ${subsection[0]}
```
Comment (sledges) [inline doc/base.md:55]: this results in extra leading space, due to untrimmed `${subsection[0]}`

Comment (vigejolla) [reply doc/base.md:55]: Fixed; now the strings are trimmed

**Pull request merged**